cobbler/items/abstract/inheritable_item.py

```diff
@@ -0,0 +1,361 @@
+"""
+"InheritableItem" the entry point for items that have logical parents and children.
+
+Changelog:
+    * V3.4.0 (unreleased):
+        * Initial creation of the class
+"""
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+# SPDX-FileCopyrightText: Enno Gotthold <enno.gotthold@suse.com>
+
+from abc import ABC
+from typing import TYPE_CHECKING, Any, Dict, List, NamedTuple, Optional, Union
+
+from cobbler.cexceptions import CX
+from cobbler.decorator import LazyProperty
+from cobbler.items.abstract.base_item import BaseItem
+
+if TYPE_CHECKING:
+    from cobbler.api import CobblerAPI
+    from cobbler.items.distro import Distro
+    from cobbler.items.menu import Menu
+    from cobbler.items.profile import Profile
+    from cobbler.items.system import System
+    from cobbler.settings import Settings
+
+
+class HierarchyItem(NamedTuple):
+    """
+    NamedTuple to display the dependency that a single item has.
+    """
+
+    dependant_item_type: str
+    dependant_type_attribute: str
+
+
+class LogicalHierarchy(NamedTuple):
+    """
+    NamedTuple to display the order of hierarchy in the dependency tree.
+    """
+
+    up: List[HierarchyItem]
+    down: List[HierarchyItem]
+
+
+class InheritableItem(BaseItem, ABC):
+    """
+    Abstract class that acts as a starting point in the inheritance for items that have parents and children.
```

Comment:
Items can't have multiple parents, right?
 
```suggestion
    Abstract class that acts as a starting point in the inheritance for items that have a parent and children.
```